x-pack/plugins/maps/public/connected_components/right_side_controls/layer_control/layer_toc/toc_entry/toc_entry_actions_popover/toc_entry_actions_popover.tsx

```diff
@@ -97,15 +153,38 @@ export class TOCEntryActionsPopover extends Component<Props, State> {
     ];
 
     if (!this.props.isReadOnly) {
+      if (this.state.isLayerEditable) {
+        actionItems.push({
+          name: EDIT_FEATURES_LABEL,
+          icon: <EuiIcon type="vector" size="m" />,
+          'data-test-subj': 'editLayerButton',
+          toolTipContent: this.state.editModeEnabled
+            ? null
+            : i18n.translate('xpack.maps.layerTocActions.editLayerTooltip', {
+                defaultMessage:
+                  'Edit features only supported for document layers without clustering, joins, or time filtering',
+              }),
+          disabled: !this.state.editModeEnabled,
+          onClick: async () => {
+            this._closePopover();
+            const supportedShapeTypes = await (this.props.layer.getSource() as ESSearchSource).getSupportedShapeTypes();
+            if (supportedShapeTypes.length === 1) {
```

Comment:
This check is not clear as to why `supportedShapeTypes.length === 1` means enablePointEditing vs `this.props.enableShapeEditing`. I know that ESSearchSource.getSupportedShapeTypes only returns VECTOR_SHAPE_TYPE.POINT for points but that assumes a lot of knowledge of ESSearchSource.getSupportedShapeTypes implemeation. As an example EmsFileSource.getSupportedShapeTypes return [VECTOR_SHAPE_TYPE.POLYGON]. That would break this logic if EmsFileSource ever started supporting feature editing (not likly but an example of where this check fails). How about changing  check to ensure line and polygon are not in list.